thingsvision/core/extraction/extractors.py

```diff
@@ -195,11 +197,49 @@ def load_model_from_source(self) -> None:
             else:
                 weights = None
             self.model = model(weights=weights)
+            prep_function_name = self.get_keras_preprocessing(self.model_name)
+            if prep_function_name is None: # we do not want to change the default preprocessing if we couldn't find a keras preprocess_input function
+                return
+
+            # first get the module name, the the preprocess_input attached to it
+            prep_function = getattr(getattr(tensorflow_models, prep_function_name), 'preprocess_input')
+
+            # different models take different sized inputs. has to be accounted for
+            resize_dim = self.model.layers[0].input_shape[0][-2] # -2 and -3 are h and w.
+            self.preprocess = tf.keras.Sequential([Lambda(prep_function), tf.keras.layers.experimental.preprocessing.Resizing(resize_dim, resize_dim)])
         else:
             raise ValueError(
                 f"\nCould not find {self.model_name} among TensorFlow models.\n"
             )
 
+    def get_keras_preprocessing(self, model_name:str) -> Union[str, None]:
+        """Get the preprocessing function for the corresponding model from `tensorflow.keras.applications.*`"""
+    
+        patterns = [
+            (r'^ConvNeXt(Base|Large|Small|Tiny|XLarge)$', 'convnext'),
+            (r'^DenseNet\d+$', 'densenet'),
+            (r'^EfficientNetB[0-7]$', 'efficientnet'),
+            (r'^EfficientNetV2(B[0-3]|[LMS])$', 'efficientnet_v2'),
+            (r'^InceptionResNetV2$', 'inception_resnet_v2'),
+            (r'^InceptionV3$', 'inception_v3'),
+            (r'^MobileNet$', 'mobilenet'),
+            (r'^MobileNetV2$', 'mobilenet_v2'),
+            (r'^MobileNetV3(Large|Small)$', 'mobilenet_v3'),
+            (r'^NasNet(Large|Mobile)$', 'nasnet'),
+            (r'^ResNet\d+$', 'resnet'),
+            (r'^ResNet\d+V2$', 'resnet_v2'),
+            (r'^VGG16$', 'vgg16'),
+            (r'^VGG19$', 'vgg19'),
+            (r'^Xception$', 'xception')
+        ]
+        # Try each pattern
+        for pattern, preprocess_value in patterns:
+            if re.match(pattern, model_name):
```

Comment:
Wouldn't omitting `$` cause unwanted matches (e.g. "MobileNet" would match "MobileNetV3" too)?

Reply:
@candemircan This was just feedback :) I am trying to understand whether there is a specific reason for `re.match` in combination with the regular expressions you used :) 


@andropar it depends on `re.match`  vs `re.search`. `re.match` searches the pattern only at the beginning of a string. Thus, `^` is baked into the function and `$` is omitted. I don't think we ever need `^` but `$` is needed for one of the two (or maybe even both the more I think about it? 🤔)

Reply:
no specific reason on my side. it was the first thing I wrote that worked correctly, and I left it as is because I don't know regexp very well. If there is a better way to do this, go ahead and change it please :)

Reply:
Ok, thanks for the clarification! I changed it from `re.match` to `re.search` because `^` is redundant with `re.match`. it's baked into the function. I left the patterns as you specified them to be as explicit as possible. Thank you for helping with this!

Reply:
coolio, thanks Lukas! all the best